beacon-chain/blockchain/service.go

```diff
@@ -90,6 +90,20 @@ func (s *Service) Start() {
 	if err != nil {
 		log.Fatalf("Could not fetch beacon state: %v", err)
 	}
+
+	if featureconfig.Get().InitSyncCacheState {
+		cp, err := s.beaconDB.FinalizedCheckpoint(ctx)
+		if err != nil {
+			log.Fatalf("Could not fetch finalized cp: %v", err)
+		}
+		if beaconState == nil {
+			beaconState, err = s.beaconDB.State(ctx, bytesutil.ToBytes32(cp.Root))
+			if err != nil {
+				log.Fatalf("Could not fetch beacon state: %v", err)
+			}
+		}
```

Comment:
Would you mind explaining what you're doing here?

Reply:
Added comment. so for running initial sync with state cache, in an event of restart, we use last finalized check point as start point to start syncing instead of head state. This is because we no longer saves state every slot.